dart/dynamics/Shape.h

```diff
@@ -191,29 +142,19 @@ class Shape : public virtual common::Subject
   /// Instruct this shape to update its data
   virtual void refreshData();
 
-  /// \brief
-  virtual void draw(renderer::RenderInterface* _ri = nullptr,
-                    const Eigen::Vector4d& _color = Eigen::Vector4d::Ones(),
-                    bool _useDefaultColor = true) const = 0;
-
-  /// Pass in true to prevent this shape from being rendered. Pass in false to
-  /// allow it to render again.
-  void setHidden(bool _hide=true);
+  /// Notify that the color (rgba) of this shape has updated
+  virtual void notifyColorUpdate(const Eigen::Vector4d& color);
```

Comment:
The VisualAddon would more or less become the key/value store. The advantage is that then an asset loader can just directly store whatever values it wants, rather than having to figure out how to map "fresnelTint" and "roughness" into "diffuse" and "specular" for example.